src/sim-host/socket.js

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
What about moving all the UI-related code to a `src/sim-host/UI/` folder, and have the consumers that want a custom UI pass in the path to their `UI/` folder instead of the `sim-host/` root? This way, we can keep all the sim-host files under `src/sim-host/` in the repo, while still making it clear that core files like this one should not be modified.

This of course assumes that we don't want to support custom protocols. I think it's perfectly fine to not allow that. Like you said, if they need such big tweaks, then it will be better for them to fork the repo.

Whatever we do, I would really prefer having everything related to sim-host be under a single `src/sim-host/` directory.